framework/src/auxkernels/TimeDerivativeAux.C

```diff
@@ -0,0 +1,75 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TimeDerivativeAux.h"
+
+registerMooseObject("MooseApp", TimeDerivativeAux);
+registerMooseObject("MooseApp", ADTimeDerivativeAux);
+
+template <bool is_ad>
+InputParameters
+TimeDerivativeAuxTempl<is_ad>::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription(
+      "Returns the time derivative of the specified variable/functor as an auxiliary variable.");
+
+  params.addCoupledVar("v", "Variable to take the time derivative of");
+  params.addParam<MooseFunctorName>("functor", "Functor to take the time derivative of");
+  params.addParam<MooseFunctorName>("factor", 1, "Factor to multiply the time derivative by");
+
+  return params;
+}
+
+template <bool is_ad>
+TimeDerivativeAuxTempl<is_ad>::TimeDerivativeAuxTempl(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _v_dot(isParamValid("v") ? &coupledDot("v") : nullptr),
+    _functor(isParamValid("functor") ? &getFunctor<GenericReal<is_ad>>("functor") : nullptr),
+    _factor(getFunctor<GenericReal<is_ad>>("factor")),
+    _use_qp_arg(dynamic_cast<MooseVariableFE<Real> *>(&_var))
+{
+  if ((!_functor && !_v_dot) || (_functor && _v_dot))
+    mooseError("A variable or a functor parameter should be provided to the "
+               "TimeDerivativeAuxTempl<is_ad>");
```

Comment:
Maybe add `name()` somewhere in here

Reply:
None of this is needed - the type and name are output with `MooseObject::mooseError()`